src/test/java/org/commcare/formplayer/tests/MenuAssertionsTest.java

```diff
@@ -0,0 +1,90 @@
+import org.commcare.formplayer.junit.Installer;
+import org.commcare.formplayer.junit.RestoreFactoryExtension;
+import org.commcare.formplayer.junit.StorageFactoryExtension;
+import org.commcare.formplayer.junit.request.Response;
+import org.commcare.formplayer.junit.request.SessionNavigationRequest;
+import org.commcare.formplayer.services.RestoreFactory;
+import org.commcare.formplayer.utils.MockRequestUtils;
+import org.commcare.formplayer.utils.TestContext;
+import org.commcare.formplayer.web.client.WebClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+
+@WebMvcTest
+@Import({MenuController.class})
+@ContextConfiguration(classes = {TestContext.class, CacheConfiguration.class})
+@ExtendWith(InitializeStaticsExtension.class)
+public class MenuAssertionsTest {
+
+    private static final String ASSERTIONS_APP = "basic";
+    @Autowired
+    protected WebClient webClientMock;
+
+    @Autowired
+    RestoreFactory restoreFactoryMock;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private MockRequestUtils mockRequest;
+
+    @RegisterExtension
+    static RestoreFactoryExtension restoreFactoryExt = new RestoreFactoryExtension.builder()
+            .withUser("user").withDomain("domain")
+            .withRestorePath("restores/basic.xml")
+            .build();
+
+    @RegisterExtension
+    static StorageFactoryExtension storageExt = new StorageFactoryExtension.builder()
+            .withUser("user").withDomain("domain").build();
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        mockRequest = new MockRequestUtils(webClientMock, restoreFactoryMock);
+    }
+
+
+    @Test
+    public void testMenuAssertions() {
+        // Check passing assertion
+        Response<CommandListResponseBean> response0 = navigate(new String[]{"0"},
+                CommandListResponseBean.class);
+        assertNotNull(response0);
+        // Check failing assertion
+        assertThrows(Exception.class, () -> {
+            navigate(new String[]{"1"},
+                CommandListResponseBean.class);
+        });
+    }
+
+
+    private <T extends BaseResponseBean> Response<T> navigate(
+            String[] selections, Class<T> responseClass) {
+        String installReference = Installer.getInstallReference(ASSERTIONS_APP);
+        SessionNavigationRequest<T> request = new SessionNavigationRequest<>(
+                mockMvc, responseClass, installReference);
+        SessionNavigationBean bean = request.getNavigationBean(selections);
+        return request.requestWithBean(bean);
```

Comment:
nit: you can simplify this a bit:
```suggestion
        return request.request(selections);
```

Reply:
simplified here https://github.com/dimagi/formplayer/pull/1411/commits/cd051b4cf20b6e5e09842c3a64c29c332c348c2d